portless/agents/node-agent/internal/agent/connect.go

```
@@ -96,18 +100,21 @@ func (r ShellRunner) cli() string {
 	return "docker"
 }
 
-func (r ShellRunner) Exec(argv []string) (string, error) {
-	if len(argv) == 0 {
-		return "", fmt.Errorf("empty argv")
+func (r ShellRunner) Execute(command Command) (string, error) {
+	if command.Path == "" || len(command.Argv) == 0 || command.Argv[0] != command.Path {
+		return "", fmt.Errorf("invalid resolved command")
 	}
-	out, err := exec.Command(argv[0], argv[1:]...).CombinedOutput()
+	out, err := exec.Command(command.Path, command.Argv[1:]...).CombinedOutput()
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)